### ISSUE TITLE
Unexpected token: NEW

### ISSUE THREAD

**Issue by bskridaila** [phpmd/phpmd#421]
Hello,

From PHP 7.0.0 such keyword - NEW is completely fine to use, as well as others are now valid to use that were forbidden/reserved by prior PHP versions from a given [list](http://php.net/manual/en/reserved.keywords.php)

currently used versions: 
phpmd - 2.5.0
pdepend - 2.3.2

**Comment by ravage84** [comment]
> As of PHP 7.0.0 these keywords are allowed as property, constant, and method names of classes, interfaces and traits, except that class may not be used as constant name. 

Honestly, I'm not sure if this is not a mess in itself. But since the language allows it, people will use it and if people will use it, PHPMD would trip over it, so it needs to support it. *sigh*

@manuelpichler am I right and this needs to be done in pdepend, instead?

**Comment by bskridaila** [comment]
Good day for all.

The problem still exists - Unexpected token: NEW
phpmd - 2.6.0
pdepend - 2.5.0

Although. It is written:
PHPMD on PDepend 2.5 which should complete support for PHP 7 language features

Thanks

**Comment by manuelpichler** [comment]
@bskridaila can you please provide an example, because I have no clue where we fobrid NEW. E.g. the following example works:

```php
<?php
class Old
{
    const NEW = 42;

    public function new() {
        echo self::NEW, PHP_EOL,  __METHOD__, PHP_EOL;
    }
}

$old = new Old();
$old->new();
```

**Comment by bskridaila** [comment]
Sure.

It seems a problem then, for a variable you try to set a value from a constant, e.g. modified example from above:

```
<?php
class Old
{
    const NEW = 42;
    private $condition = self::NEW;

    public function new()
    {
        echo  $this->condition, PHP_EOL, __METHOD__, PHP_EOL;
    }
}
```

or another case, just tried:

`const PREVIOUS_STATUS = Old::NEW; // this also will show the same error regarding NEW`

Thanks

**Comment by t-geindre** [comment]
Hi,

I encoutered the same error with a class constant named `PRIVATE` and I guess it is the same bug as with `NEW`.

Here is a simple code example:

```php
<?php
class Foo
{
    const PRIVATE = 'my-private-const';

    function bar(string $foo = self::PRIVATE)
    {
    }
}
```

I run the following command:
```shell
$ ./vendor/bin/phpmd test.php text unusedcode
```

Then I get the following error:
> test.php - Unexpected token: PRIVATE, line: 6, col: 38, file: test.php.

PHPMD - 2.6.0
PDEPEND - 2.5.2

**Comment by kylekatarnls** [comment]
Hi, this has been fixed in Hi, this is fixed in [pdepend/pdepend@2.6.0-beta.1](https://github.com/pdepend/pdepend/releases/tag/2.6.0-beta.1)

Note that last PHPMD.phar release does not yet contain this version of PDepend but you already can test it using `composer` (using `"beta"` as minimum-stability).
